File: src/Heart.cpp
#include "Animation.h"
#include "Character.h"
#include "Heart.h"
#include "resources.h"


Heart::Heart() {
  animation = new Animation();
  animation->addFrame(IMG("heart.bmp"));
}


void Heart::collectedBy(Character *character) {
  character->setHealth(character->getHealth() + 1);
  play_sound(SND("heart.wav"));
}

File: src/Heart.m
#import "Heart.h"


@implementation Heart


- init {
  
  self = [super init];
  
  if (self) {
    animation = [[Animation alloc] init];
    [animation addFrame: getImage(IMAGES_HEART)];
  }
  
  return self;
  
}


- collectedBy: (id<Livable, Moneyable>) character {
  [character setHealth: [character getHealth] + 1];
  playSound(SOUNDS_HEART);
  return self;
}


@end

